src/nxt_socket.h

```diff
@@ -92,15 +103,15 @@ typedef union {
 
 
 NXT_EXPORT nxt_socket_t nxt_socket_create(nxt_task_t *task, nxt_uint_t family,
-    nxt_uint_t type, nxt_uint_t protocol, nxt_uint_t flags);
+    nxt_uint_t type, nxt_uint_t protocol, nxt_uint_t flags, nxt_err_t *errnum);
 NXT_EXPORT void nxt_socket_defer_accept(nxt_task_t *task, nxt_socket_t s,
     nxt_sockaddr_t *sa);
 NXT_EXPORT nxt_int_t nxt_socket_getsockopt(nxt_task_t *task, nxt_socket_t s,
     nxt_uint_t level, nxt_uint_t sockopt);
 NXT_EXPORT nxt_int_t nxt_socket_setsockopt(nxt_task_t *task, nxt_socket_t s,
     nxt_uint_t level, nxt_uint_t sockopt, int val);
 NXT_EXPORT nxt_int_t nxt_socket_bind(nxt_task_t *task, nxt_socket_t s,
```

Comment:
Note you can use `nxt_socket_errno` after calling `nxt_socket_bind()`. So it's not a good idea to add one more parameter `*errnum`. It's better to keep the API as minimal as possible, including its parameters.

Reply:
I did that before. But what worries me is that virtually any other function call (like nxt_alert and others) could theoretically cause errno to change its state. I think this is probably why nxt_listen_socket_create uses pure listen and connect for errno.

Reply:
I mean (see "Notes"):
https://linux.die.net/man/3/errno

Reply:
@hongzhidao what do you think about the above described by me?

Reply:
It makes sense, but the decision of where to do `listen` affects such changes.

Reply:
I want to draw your attention to the fact that in this discussion we are discussing `errno`.

Reply:
@hongzhidao Let's get this straight. Is it still safe to call `nxt_socket_errno` after `nxt_alert` and other calls?